File: core/src/com/spacefist/components/Sound.java
package com.spacefist.components;

import com.spacefist.GameData;
import com.spacefist.components.abst.SoundComponent;
import com.spacefist.entities.Entity;

/**
 * A simple sound component that plays a sound effect.
 */
public class Sound implements SoundComponent {
    com.badlogic.gdx.audio.Sound soundEffect;

    /**
     * Creates a new Sound instance from a SoundEffect.
     *
     * @param sound The sound to play
     */
    public Sound(com.badlogic.gdx.audio.Sound sound) {
        this.soundEffect = sound;
    }

    /**
     * Plays the sound.
     */
    public void play() {
        soundEffect.play();
    }

    public void update(GameData gameData, Entity obj) { }
}

File: core/src/com/spacefist/components/Sound.kt
package com.spacefist.components

import com.spacefist.GameData
import com.spacefist.components.abst.SoundComponent
import com.spacefist.entities.Entity

/**
 * A simple sound component that plays a sound effect.
 */
class Sound
/**
 * Creates a new Sound instance from a SoundEffect.
 *
 * @param sound The sound to play
 */
(internal var soundEffect: com.badlogic.gdx.audio.Sound) : SoundComponent {

    /**
     * Plays the sound.
     */
    fun play() {
        soundEffect.play()
    }

    override fun update(gameData: GameData, obj: Entity) {}
}